doc/filters/country_name.rst

```diff
@@ -16,6 +16,32 @@ By default, the filter uses the current locale. You can pass it explicitly:
     {# États-Unis #}
     {{ 'US'|country_name('fr') }}
 
+The locale can contain more than two letters depending on the region:
+
+.. code-block:: twig
+
+    {# 美國 #}
+    {{ 'US'|country_name('zh_Hant_HK') }}
+
+.. note::
+
+    For more information on the format of the locale:
+
+    See https://www.rfc-editor.org/info/bcp47 for the specifications.
+
+    It is documented by https://www.php.net/manual/en/class.locale.php.
+
+    Taking the previous example into consideration, the filter searches for the 'US' country key within the array located
+    in the 'zh_Hant_HK.php' file corresponding to the locale and then retrieves the associated country name.
+
+If the specified locale were to be unknown, it will default to the closest available locale instead:
```

Comment:
I don't think we need this info. The fact that the CLDR data has a concept of fallback is quite an internal matter anyway (even `fr_FR` might actually read a name from the `fr` data if there is no override for it in `fr_FR`). This is not about being known or no.